File: csgo/surf/bulkDownloadSH.js
let fs = require("fs");
let http = require("http");

let rawdata = fs.readFileSync('maplist.json'); // - file format {maps:[Array]}
let maps = JSON.parse(rawdata).maps;

maps.forEach(map => {
	console.log(map + "...");
    let file = fs.createWriteStream(map + ".bsp.bz2");
    let request = http.get("http://fast.surfdl.net/csgo/maps/"+ map +".bsp.bz2", function(response) {
        if (response) response.pipe(file);
        else console.log("failed!");
    });
});

File: csgo/surf/bulkDownloadSH.py
import os
import urllib.request
dir, file = os.path.split(os.path.abspath(__file__))

maps = urllib.request.urlopen("https://pastebin.com/raw/xxxxxxxx") # Maplist - separator \n
maps = maps.read()
maps = maps.splitlines()

fastdl = "http://fast.surfdl.net/csgo/maps/"

for map in maps:
    map = map.decode("utf-8")
    print("Downloading", map)
    url = fastdl + map + ".bsp.bz2"
    dldir = dir + "\\" + map + ".bsp.bz2"
    try:
        urllib.request.urlretrieve(url, dldir)
        print("Downloaded", map)
    except:
        print("Failed to download", url)
        try:
            url = fastdl + map + ".bsp"
            dldir = dir + "\\" + map + ".bsp"
            urllib.request.urlretrieve(url, dldir)
            print("Downloaded", map)
        except:
            print("Failed to download", url)